Tests/SwiftLintFrameworkTests/ConfigurationTests+MultipleConfigs.swift

```diff
@@ -411,12 +411,159 @@ extension ConfigurationTests {
                 disabled: testCase.disabledInChild ? [ruleIdentifier] : [],
                 optIn: testCase.optedInInChild ? [ruleIdentifier] : []
             ))
-            let mergedConfiguration = parentConfiguration.merged(withChild: childConfiguration, rootDirectory: "")
+            let mergedConfiguration = parentConfiguration.merged(withChild: childConfiguration)
             let isEnabled = mergedConfiguration.contains(rule: ruleType)
             XCTAssertEqual(isEnabled, testCase.isEnabled, testCase.message)
         }
     }
 
+    // MARK: Warnings about configurations for disabled rules
+    func testDefaultConfigurationDisabledRuleWarnings() {
+        let optInRuleType = ImplicitReturnRule.self
+        XCTAssertTrue((optInRuleType as Any) is any OptInRule.Type)
+        testDefaultConfigurationDisabledRuleWarnings(for: optInRuleType)
+
+        let defaultRuleType = BlockBasedKVORule.self
+        XCTAssertFalse((defaultRuleType as Any) is any OptInRule.Type)
+        testDefaultConfigurationDisabledRuleWarnings(for: defaultRuleType)
+    }
+
+    private func testDefaultConfigurationDisabledRuleWarnings(for ruleType: any Rule.Type) {
+        let ruleIdentifier = ruleType.identifier
+
+        let parentConfigurations = [
+            nil,
+            Configuration.emptyDefaultConfiguration(),
+            Configuration.optInDefaultConfiguration(ruleIdentifier),
+            Configuration.optInDisabledDefaultConfiguration(ruleIdentifier),
+            Configuration.disabledDefaultConfiguration(ruleIdentifier),
+            Configuration.emptyOnlyConfiguration(),
+            Configuration.enabledOnlyConfiguration(ruleIdentifier),
+            Configuration.allEnabledConfiguration()
+        ]
+
+        let configurations = [
+            Configuration(rulesMode: .default(disabled: [], optIn: [])),
+            Configuration(rulesMode: .default(disabled: [], optIn: [ruleIdentifier])),
+            Configuration(rulesMode: .default(disabled: [ruleIdentifier], optIn: [ruleIdentifier])),
+            Configuration(rulesMode: .default(disabled: [ruleIdentifier], optIn: []))
+        ]
+
+        for parentConfiguration in parentConfigurations {
+            testParentConfiguration(parentConfiguration, configurations: configurations, ruleType: ruleType)
```

Comment:
Move the loop over `configurations` here to indicate that all combinations are checked here.